### PR TITLE
Refactor: project architecture

### PR DESCRIPTION
**Pull request by helderberto**
This PR contains:

- feat: All libs upgrade to latest stable version;
- feat: `.npmrc` config to save-exact and doesn't save package-lock;
- refactor: sources moved to `src` directory;
- refactor: tests files moved to `__tests__`;
- refactor: examples moved to `docs`.

@jamiebuilds could u check this PR? :)

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+package-lock = false
+save-exact = true
```
Comment (hg-pyun) [inline .npmrc]: Need EOF.

Comment (helderberto) [reply .npmrc]: Fixed :)

**Comment by helderberto** [comment]
That's okay now? @hg-pyun

**Review by hg-pyun — APPROVED** [review]
Looks good.

**Pull request closed without merging**